crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/service/MagsProceedingService.java

```diff
@@ -0,0 +1,81 @@
+package uk.gov.justice.laa.crime.crowncourt.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.justice.laa.crime.crowncourt.builder.UpdateRepOrderDTOBuilder;
+import uk.gov.justice.laa.crime.crowncourt.dto.CrownCourtDTO;
+import uk.gov.justice.laa.crime.crowncourt.dto.maatcourtdata.RepOrderDTO;
+import uk.gov.justice.laa.crime.crowncourt.model.MagsDecisionResult;
+import uk.gov.justice.laa.crime.crowncourt.model.common.ApiFinancialAssessment;
+import uk.gov.justice.laa.crime.crowncourt.model.common.ApiHardshipOverview;
+import uk.gov.justice.laa.crime.crowncourt.model.common.ApiIOJSummary;
+import uk.gov.justice.laa.crime.enums.*;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.Optional.ofNullable;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class MagsProceedingService {
+
+    private final MaatCourtDataService maatCourtDataService;
+
+    private final Set<CaseType> magsCourtCaseTypes = Set.of(
+            CaseType.INDICTABLE, CaseType.EITHER_WAY, CaseType.SUMMARY_ONLY
+    );
+
+    public MagsDecisionResult determineMagsRepDecision(CrownCourtDTO dto) {
+        ApiIOJSummary iojSummary = dto.getIojSummary();
+        ReviewResult iojResult = ReviewResult.getFrom(
+                iojSummary.getDecisionResult() != null
+                        ? iojSummary.getDecisionResult() : iojSummary.getIojResult()
+        );
+        boolean isMagsCourt = magsCourtCaseTypes.contains(dto.getCaseType());
+        if (iojResult != null && isMagsCourt) {
+            DecisionReason decisionReason = getDecisionReason(dto, iojResult);
+            if (decisionReason != null) {
+                MagsDecisionResult decisionResult =
+                        MagsDecisionResult.builder()
+                                .decisionDate(LocalDate.now())
+                                .decisionReason(decisionReason)
+                                .build();
+                dto.setMagsDecisionResult(decisionResult);
+                RepOrderDTO repOrderDTO =
+                        maatCourtDataService.updateRepOrder(UpdateRepOrderDTOBuilder.build(dto, decisionResult));
```

Comment:
In the stored procedure update_application_mags_dec before updating the repOrder with the MagDecision, pre_update_checks function is invoked - can this be skipped?
After pre_update_checks we have: 
   handle_eform_result(p_app_obj => p_application_object, p_invoker => 'update_application_mags_dec');
Should we call ATS service from Orchestration after calling determineMagsRepDecision?

Reply:
Yes, I had assumed we would call this in the Orchestration service. Looking at the procedure, it doesn't look like there is a problem if we want to change the ordering of the calls. I've gone back and double checked and I'm confident that we don't need the call to `pre_update_checks`.